src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -90,13 +227,43 @@ QJSValue ControllerScriptEngineLegacy::wrapFunctionCode(
     return wrappedFunction;
 }
 
+#ifdef MIXXX_USE_QML
+void ControllerScriptEngineLegacy::setModulePaths(
+        const QList<LegacyControllerMapping::QMLModuleInfo>& modules) {
+    const QStringList paths = m_fileWatcher.files();
+    if (!paths.isEmpty()) {
+        m_fileWatcher.removePaths(paths);
+    }
+
+    m_modules = modules;
+}
+void ControllerScriptEngineLegacy::setInfoScreens(
+        const QList<LegacyControllerMapping::ScreenInfo>& screens) {
+    m_rootItems.clear();
+    m_renderingScreens.clear();
+    m_transformScreenFrameFunctions.clear();
+    m_infoScreens = screens;
+}
+#endif
+
 void ControllerScriptEngineLegacy::setScriptFiles(
         const QList<LegacyControllerMapping::ScriptFileInfo>& scripts) {
     const QStringList paths = m_fileWatcher.files();
     if (!paths.isEmpty()) {
         m_fileWatcher.removePaths(paths);
     }
     m_scriptFiles = scripts;
+
+#ifdef MIXXX_USE_QML
+    for (const LegacyControllerMapping::ScriptFileInfo& script : std::as_const(m_scriptFiles)) {
+        if (script.type == LegacyControllerMapping::ScriptFileInfo::Type::Qml) {
+            setQMLMode(true);
+            return;
+        }
+    }
+
+    setQMLMode(false);
```

Comment:
std::algorithms ftw, and we might as well allow it to be `par_unseq`, could make a difference with many files.
```suggestion
    setQMLMode(std::any_of(std::execution::par_unseq, m_scriptFiles.cbegin(), m_scriptFiles.cend(), [](const auto& scriptFileInfo){
        return scriptFileInfo.type == LegacyControllerMapping::ScriptFileInfo::Type::Qml);
    }));
```

Reply:
Looks like the following did the trick
```c++
#pragma push_macro("emit")
#undef emit
#include <execution>
#pragma pop_macro("emit")
```

Reply:
yikes. thats also ugly. I guess then lets ignore the parallel algorithm for now. 

Whats the error? what compiler/stdlib? I can't reproduce the issue on compiler explorer.